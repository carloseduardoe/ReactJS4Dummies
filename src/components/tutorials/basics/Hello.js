import React from 'react'

const Hello = ({name}) => {
  return (
    <div data-testid="hello_container" className="dummyClass">
      <h6 data-testid="hello_message">Hi {name}!</h6>
    </div>
  )
}

export default Hello
