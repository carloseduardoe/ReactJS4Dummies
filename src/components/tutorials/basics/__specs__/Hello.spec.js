import React from 'react';
import ReactDOM from 'react-dom';
import Hello from '../Hello';

import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('renders', () => {
    const div = document.createElement('div');
    const hello = <Hello name={name}/>;
    ReactDOM.render(hello, div);
});

it('renders correctly', () => {
    const name = 'Carlos';
    const hello = <Hello name={name}/>;
    const { getByTestId } = render(hello);

    expect(getByTestId('hello_container')).toHaveClass('dummyClass');
    expect(getByTestId('hello_message')).toHaveTextContent(`Hi ${name}!`);
});

it('matches snapshot', () => {
    const name = 'Carlos';
    const tree = renderer.create(<Hello name={name}/>).toJSON();

    expect(tree).toMatchSnapshot();
});