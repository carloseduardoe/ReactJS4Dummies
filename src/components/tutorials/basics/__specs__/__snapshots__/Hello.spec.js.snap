// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello matches snapshot 1`] = `
<div
  className="hello-component_container"
  data-testid="hello-component_container"
>
  <h6
    data-testid="hello-component_message"
  >
    Hi 
    Carlos
    !
  </h6>
</div>
`;
